core/src/data_sources/qdrant.rs

```diff
@@ -152,3 +179,282 @@ impl QdrantClients {
         }
     }
 }
+
+#[derive(Clone)]
+pub struct DustQdrantClient {
+    client: Arc<QdrantClient>,
+    pub cluster: QdrantCluster,
+}
+
+impl DustQdrantClient {
```

Comment:
Any reason of not having two distinct implementation (one for v0 and one for v1)? The match clutters a lot each method ...

Reply:
We can definitely but really the goal is to delurk v0 once we're done so I didn't want to invest too much in this since it's all meant to disapear.

Reply:
Also more code

Reply:
And you would  have to delurk the routing etc...